dotnet/src/Experimental/Assistants/Internal/Assistant.cs

```diff
@@ -74,61 +67,73 @@ internal sealed class Assistant : IAssistant
+                .WithOpenAIChatCompletion(this._model.Model, this._restContext.ApiKey);
+
+        if (plugins is not null)
+        {
+            builder.ConfigurePlugins(
+                c =>
+                {
+                    foreach (var plugin in plugins)
+                    {
+                        c.Add(plugin);
+                    }
+                });
+        }
+
+        this.Kernel = builder.Build();
     }
 
-    /// <inheritdoc/>
-    public Task<IChatThread> NewThreadAsync(CancellationToken cancellationToken = default)
-    {
-        return ChatThread.CreateAsync(this._restContext, cancellationToken);
-    }
+/// <inheritdoc/>
+public Task<IChatThread> NewThreadAsync(CancellationToken cancellationToken = default)
+{
+    return ChatThread.CreateAsync(this._restContext, cancellationToken);
+}
 
-    /// <inheritdoc/>
-    public Task<IChatThread> GetThreadAsync(string id, CancellationToken cancellationToken = default)
-    {
-        return ChatThread.GetAsync(this._restContext, id, cancellationToken);
-    }
+/// <inheritdoc/>
+public Task<IChatThread> GetThreadAsync(string id, CancellationToken cancellationToken = default)
+{
+    return ChatThread.GetAsync(this._restContext, id, cancellationToken);
+}
 
-    /// <summary>
-    /// Marshal thread run through <see cref="KernelFunction"/> interface.
-    /// </summary>
-    /// <param name="input">The user input</param>
-    /// <param name="cancellationToken">A cancellation token.</param>
-    /// <returns>An assistant response (<see cref="AssistantResponse"/></returns>
-    [KernelFunction, Description("Provide input to assistant a response")]
-    public async Task<string> AskAsync(
-        [Description("The input for the assistant.")]
+/// <summary>
+/// Marshal thread run through <see cref="KernelFunction"/> interface.
+/// </summary>
+/// <param name="input">The user input</param>
+/// <param name="cancellationToken">A cancellation token.</param>
+/// <returns>An assistant response (<see cref="AssistantResponse"/></returns>
+[KernelFunction, Description("Provide input to assistant a response")]
+public async Task<AssistantResponse> AskAsync(
+    [Description("The input for the assistant.")]
         string input,
-        CancellationToken cancellationToken = default)
-    {
-        var thread = await this.NewThreadAsync(cancellationToken).ConfigureAwait(false);
-        await thread.AddUserMessageAsync(input, cancellationToken).ConfigureAwait(false);
-        var message = await thread.InvokeAsync(this, cancellationToken).ConfigureAwait(false);
-        var response =
-            new AssistantResponse
-            {
-                ThreadId = thread.Id,
-                Response = string.Concat(message.Select(m => m.Content)),
-            };
-
-        return JsonSerializer.Serialize(response);
-    }
+    CancellationToken cancellationToken = default)
+{
+    var thread = await this.NewThreadAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
Won't this leave a ton of abandoned threads around?  If this is creating a thread and using it once, should it also delete the thread?

Reply:
All of these object are intended to persist.   Good question, but not affected by this priority fix.